### PR TITLE
Implement caddy instances section

### PR DESCRIPTION
**Pull request by ArtemStepanov**
Implement the complete 'Instances' section to enable comprehensive management of Caddy server instances with a rich UI.

This PR introduces the full Instances dashboard, including grid and table views, add/edit/delete/test connection dialogs, filtering, sorting, empty states, loading skeletons, and robust form validation, all adhering to the detailed technical specification.

---
<a href="https://cursor.com/background-agent?bcId=bc-af369f43-2ce6-4ea6-9842-d4ca21b2c6cc"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">Open in Cursor</picture></a>&nbsp;<a href="https://cursor.com/agents?id=bc-af369f43-2ce6-4ea6-9842-d4ca21b2c6cc"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">Open in Web</picture></a>

### CODE REVIEW

**Comment by ArtemStepanov** [comment]
@cursor refresh instance status after a new instance has been added.

File: src/components/instances/DeleteInstanceDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "@/components/ui/alert-dialog";
+import { Button } from "@/components/ui/button";
+import { Checkbox } from "@/components/ui/checkbox";
+import { Label } from "@/components/ui/label";
+import { CaddyInstance } from "@/lib/api-client";
+import { mapInstanceStatus } from "@/lib/instance-utils";
+
+interface DeleteInstanceDialogProps {
```
Comment (ArtemStepanov) [inline src/components/instances/DeleteInstanceDialog.tsx]: @cursor let's move all models to a separate designated package in all components
